org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/BasicAuthAsyncHttpClientManager.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.shared.http.async;
+
+import org.apache.hc.client5.http.auth.UsernamePasswordCredentials;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClientBuilder;
+
+public class BasicAuthAsyncHttpClientManager extends DefaultAsyncHttpClientManager {
+
+  private String m_user;
+  private String m_password;
+
+  public BasicAuthAsyncHttpClientManager withUser(String user) {
+    m_user = user;
+    return this;
+  }
+
+  public BasicAuthAsyncHttpClientManager withPassword(String password) {
+    m_password = password;
+    return this;
+  }
+
+  @Override
+  protected void interceptCreateClient(HttpAsyncClientBuilder builder) {
+    UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(m_user, m_password.toCharArray());
+    builder.setDefaultCredentialsProvider((authScope, context) -> credentials);
```

Comment:
That does not work the same way as org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/BasicAuthHttpTransportManager.java; previously always Authorization header was set preemptively for every request/host.